Fix timeout options at daemon invokation time. This option was not correctly
used because it wasn't correctly parsed by autofs init script.
If no timeout is defined into auto.master, 10 minutes will be set by default.

Signed-off-by: Giuseppe Condorelli <giuseppe.condorelli@st.com>
--- autofs-3.1.7/samples/rc.autofs.in.orig	2010-11-15 10:12:38.000000000 +0100
+++ autofs-3.1.7/samples/rc.autofs.in	2010-11-15 10:15:32.000000000 +0100
@@ -82,13 +82,17 @@ then
 			-a x`echo "$map" | cut -c1` != 'x-' ]
 	    then
 		map=`echo "/etc/$map" | sed -e 's:^/etc//:/:'`
-		options=`echo "$options" | sed -e 's/\(^\|[ \t]\)-/\1/g'`
+		options=`echo "$options" | sed -e 's/\(^\|[ \t]\)--timeout=/\1/g'`
+		if [ -z "$options" ]
+		then
+			options=`echo 600`
+		fi
 		if [ -x $map ]; then
-		    echo "$DAEMON $dir program $map $options $localoptions"
+		    echo "$DAEMON -t $options $dir program $map $localoptions"
 		elif [ -f $map ]; then
-		    echo "$DAEMON $dir file $map $options $localoptions"
+		    echo "$DAEMON -t $options $dir file $map $localoptions"
 		else
-		    echo "$DAEMON $dir `basename $map` $options $localoptions"
+		    echo "$DAEMON -t $options $dir `basename $map` $localoptions"
 		fi
 	    fi
 	done
@@ -206,12 +210,12 @@ function debian()
 case "$1" in
     start)
 	echo -n 'Starting automounter:'
-	getmounts | while read cmd mnt rest
+	getmounts | while read cmd t_opt t_value mnt rest
 	do
 		echo -n " $mnt"
 		pidfile=/var/run/autofs`echo $mnt | sed 's/\//./'`.pid
 		start-stop-daemon --start --pidfile $pidfile --quiet \
-			--exec $DAEMON -- $mnt $rest
+			--exec $DAEMON -- -t $t_value $mnt $rest
 		#
 		#	Automount needs a '--pidfile' or '-p' option.
 		#	For now we look for the pid ourself.
